source/slides/02_linux_basics.rst

```diff
@@ -0,0 +1,507 @@
+
+  * GNU sells printed manuals as well
+
+* Emacs Wiki (`https://www.emacswiki.org`)
+* Emacs Tutorial (``C-h t`` inside emacs)
+
+Editor questions?
+-----------------
+
+|
+
+* Open an editor, find a cheat sheet, try to add some text
+* Modify the text: "``disemvowel``" it
+
+.. code-block:: bash
+
+    $ vim testvim.txt            $ emacs testemacs.txt
+    <i>                          Hello world!
+    Hello world!                 <alt + x>
+    <esc>                        replace-regexp
+    :%s/[aeiou]//g               <enter>
+    <enter>                      [aeiou]
+    :wq                          <enter>
+    <enter>                      <ctrl + x> <ctrl + s>
+                                 <ctrl + x> <ctrl + c>
+
+Git
+===
+
+Git
+---
+
+.. figure:: ../_static/Linus_Torvalds.jpeg
+    :align: left
+
+git, noun. Brit.informal.
+1. an unpleasant or contemptible person.
+
+Setting up Git
+--------------
+
+* In VM:
+
+.. code-block:: bash
+
+    $ sudo yum install git
+    $ git config --global user.name "My Name"
+    $ git config --global user.email "myself@gmail.com"
+    $ git config --global core.editor "emacs"
+    $ git config --global push.default "current" # Or "nothing"
+
+Using Git Locally
+-----------------
+
+``$ git init``
+
+.. note:: This initializes a git repo. Use `man git-init` for more info.
+
+``$ git add <filename>``
+
+.. note:: This puts <filename> into the staging area. It isn't committed yet.
+    Use ``git diff`` to see what changes aren't yet in staging.
+
+``$ git commit -m "I did a thing!"``
+
+.. note:: This actually makes the commit. Use ``git status`` to see what's in
+    staging but not yet committed. Use ``git show`` or ``git log`` to see
+    recent commits.
+
+* Undo things?
+  the `git book <http://git-scm.com/book/en/Git-Basics-Undoing-Things>`_ explains
+  well
+
+* Did I remember to commit that?::
+    
+    $ git status
+
+* What commits have I made lately?::
+
+    $ git log
+
+What Not To Do
+--------------
+
+* Don't delete the .git files
+
+.. note:: If you kill them, git loses its memory :(
+
+* Avoid redundant copies of the same work in one revision
+* Don't make "oops, undoing that" commits.
+    * Use git commit --amend or git revert
+
+.. note:: Amending is fine as long as you haven't pushed yet. It's generally a
+    bad idea to amend or rebase work that you've already shared with others,
+    unless you really know what you're doing.
+
+* Don't wait too long between commits
+    * You can squash them all together later
+
+.. note:: Commit every time you think you might want to return to the current 
+    state. You can revert back to any previous commit, but there is no way to
+    magically add a commit in where you forgot to make one.
+
+* Don't commit secrets...
+
+* If you do commit secrets, `reset them`_
+
+.. _reset them: https://jordan-wright.github.io/blog/2014/12/30/why-deleting-sensitive-information-from-github-doesnt-save-you/
```

Comment:
The link I gave is actually a response to that blog post (it links to it) explaining how the bots listen to the github stream :)